presto-main/src/test/java/io/prestosql/sql/planner/TestLocalExecutionPlanner.java

```diff
@@ -56,6 +56,7 @@ public void testCompilerFailure()
     private void assertFails(@Language("SQL") String sql, ErrorCodeSupplier supplier)
     {
         assertPrestoExceptionThrownBy(() -> runner.execute(sql))
-                .hasErrorCode(supplier);
+                .hasErrorCode(supplier).hasMessage("Query exceeded maximum columns or filters. " +
```

Comment:
We can shorten this a bit by only checking the start of the message. Also, we would prefer to format each of the checks on separate lines for readability
```java
.hasErrorCode(supplier)
.hasMessageStartingWith("Query exceeded maximum columns or filters")

Reply:
Instead of handing in `LocalExecutionPlanner` why can't we handle these compilation errors in the corresponding compiler class(`ExpressionCompiler` and `PageFunctionCompiler`). We might be able to drill down the exact problem also like whether it might be bcoz of filters or projections.

Reply:
Good idea, made that change